Sources/Sync/SyncCoordinator/DTO/Campaign.swift

```diff
@@ -0,0 +1,397 @@
+//
+//  Campaign.swift
+//  RoverSync
+//
+//  Created by Andrew Clunis on 2019-01-21.
+//  Copyright © 2019 Rover Labs Inc. All rights reserved.
+//
+
+import Foundation
+
+protocol Predicate {
+    
+}
+
+enum PredicateType : Decodable {
+    case comparisonPredicate
+    case compoundPredicate
+
+    enum CodingKeys: String, CodingKey {
+        case typeName = "__typename"
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let typeName = try container.decode(String.self, forKey: .typeName)
+        switch typeName {
+        case "ComparisonPredicate":
+            self = .comparisonPredicate
+        case "CompoundPredicate":
+            self = .compoundPredicate
+        default:
+            throw DecodingError.dataCorruptedError(forKey: CodingKeys.typeName, in: container, debugDescription: "Expected either ComparisonPredicate or CompoundPredicate – found \(typeName)")
+        }
+    }
+}
+
+enum ComparisonPredicateModifier : String, Decodable {
+    case direct = "DIRECT"
+    case any = "ANY"
+    case all = "ALL"
+}
+
+enum ComparisonPredicateOperator : String, Decodable {
+    case lessThan = "LESS_THAN"
+    case lessThanOrEqualTo = "LESS_THAN_OR_EQUAL_TO"
+    case greaterThan = "GREATER_THAN"
+    case greaterThanOrEqualTo = "GREATER_THAN_OR_EQUAL_TO"
+    case equalTo = "EQUAL_TO"
+    case notEqualTo = "NOT_EQUAL_TO"
+    case like = "LIKE"
+    case beginsWith = "BEGINS_WITH"
+    case endsWith = "ENDS_WITH"
+    case in_ = "IN"
+    case contains = "CONTAINS"
+    case between = "BETWEEN"
+    case geoWithin = "GEO_WITHIN"
+}
+
+enum CompoundPredicateLogicalType : String, Decodable {
+    case or = "OR"
+    case and = "AND"
+    case not = "NOT"
+}
+
+struct ComparisonPredicate : Predicate, Decodable {
+    let keyPath: String
+    let modifier: ComparisonPredicateModifier
+    let op: ComparisonPredicateModifier
```

Comment:
Consider using backticks to get around the reserved word issue. Also, is `ComparisonPredicateModifier` the right type for this?